erdjs/src/dapp/walletConnectProvider.ts

```diff
@@ -0,0 +1,196 @@
+    };
+
+    /**
+     * Mocks a logout request by returning true
+     */
+    async logout(): Promise<boolean> {
+        if (!this.walletConnector) {
+            Logger.error("logout: Wallet Connect not initialised, call init() first");
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+        this.walletConnector.killSession();
+        return true;
+    }
+
+    /**
+     * Fetches current selected ledger address
+     */
+    async getAddress(): Promise<string> {
+        return this.address;
+    }
+
+    /**
+     * Signs and sends a transaction. Returns the transaction hash
+     * @param transaction
+     */
+    async sendTransaction(transaction: Transaction): Promise<Transaction> {
+        if (!this.walletConnector) {
+            Logger.error("sendTransaction: Wallet Connect not initialised, call init() first");
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+
+        const address = await this.getAddress();
+        const sig = await this.signTransaction("erd_sign", this.prepareWalletConnectMessage(transaction, address));
+        if (sig && sig.signature) transaction.applySignature(new Signature(sig.signature), new Address(address));
+
+        await transaction.send(this.provider);
+
+        return transaction;
+    }
+
+    private async signTransaction(method: any, params: any): Promise<any> {
+        if (!this.walletConnector) {
+            Logger.error("signTransaction: Wallet Connect not initialised, call init() first");
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+        if (this.walletConnector.connected) {
+            const response = await this.walletConnector.sendCustomRequest({
+                method,
+                params,
+            });
+
+            return response;
+        } else {
+            return false;
+        }
+    }
+
+    private prepareWalletConnectMessage(transaction: Transaction, address: string): any {
+        return {
+            nonce: transaction.getNonce().valueOf(),
+            from: address,
+            to: transaction.getReceiver().toString(),
+            amount: transaction.getValue().toString(),
+            gasPrice: transaction
+                .getGasPrice()
+                .valueOf()
+                .toString(),
+            gasLimit: transaction
+                .getGasLimit()
+                .valueOf()
+                .toString(),
+            data: Buffer.from(
+                transaction
+                    .getData()
+                    .toString()
+                    .trim()
+            ).toString(),
+            chainId: transaction.getChainID().valueOf(),
+            version: transaction.getVersion().valueOf(),
+        };
+    }
+
+    private addressIsValid(destinationAddress: string): boolean {
+        try {
+            new Address(destinationAddress);
```

Comment:
Additional check is needed here. Because inside the `Address` constructor we have something like this 

```js
if (!value) {
    return;
}
```

```js
const addr = new Address(destinationAddress);
return !!addr;
```

Reply:
Done